site/pages/admin/social-media-accounts.page.tsx

```diff
@@ -1,51 +1,180 @@
+                value: token.Value,
+                name: token.Name,
+                userId: token?.Name?.split("hootsuite/")[1]?.split("-")[0] ?? "",
+                accountType: startCase(token?.Name?.split("/")[3]) ?? "",
+                addedBy: token?.Name?.split("/")[4]?.replace("_", " ")?.split("-")[1] ?? "",
+            };
+        });
+
+        if (refreshTokens && refreshTokens.length > 0) {
+            await Promise.all(
+                refreshTokens?.map(async (token) => {
+                    const resp = await fetch(`${HOOTSUITE_URL}oauth2/token`, {
+                        method: "POST",
+                        body: new URLSearchParams({
+                            grant_type: "refresh_token",
+                            refresh_token: token.value ?? "",
+                        }),
+                        headers: {
+                            "Content-Type": "application/x-www-form-urlencoded",
+                            Authorization: authToken,
+                        },
+                    });
+                    if (resp.ok) {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        const tokenResult: HootsuiteToken = await resp.json();
+                        const keys = await getParametersByPath(`/social/${orgId}/hootsuite`);
+
+                        if (!keys || (refreshTokens && keys.Parameters?.length === 0)) {
+                            throw new Error("Refresh token is required to fetch dropdown data");
+                        }
+                        const key: string =
+                            keys.Parameters?.find((rt) => rt.Name?.includes(`${token.userId}`))?.Name || "";
+                        if (!key) {
+                            throw new Error("Refresh token is required to fetch dropdown data");
+                        }
+
+                        await putParameter(key, tokenResult.refresh_token ?? "", "SecureString", true);
+                        const userDetailsResponse = await fetch(`${HOOTSUITE_URL}v1/me`, {
+                            method: "GET",
+                            headers: {
+                                Authorization: `Bearer ${tokenResult.access_token ?? ""}`,
+                            },
+                        });
+                        if (userDetailsResponse.ok) {
+                            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                            const userDetails: HootsuiteMe = await userDetailsResponse.json();
+                            const userInfo = userDetails.data || {};
+
+                            const extraInfo = {
+                                ...userInfo,
+                                accountType: token.accountType || "",
+                                addedBy: token.addedBy || "",
+                                expiresIn: "Never",
+                            } as SocialMediaAccountsSchema[0];
```

Comment:
Type assertion might not be required if `userDetails` is parsed using zod

Reply:
done